src/endpoints/ECatCourse/index.ts

```diff
@@ -744,6 +756,384 @@ class ECatCourse extends EndpointCategory {
+      const sourceId = sourceAssignmentGroupIds[i];
+      const sourceAssignmentGroup = await this.api.course.assignmentGroup.get({
+        assignmentGroupId: sourceId,
+        courseId: sourceCourseId,
+      });
+      // check if apply_assignment_group_weights is true in the source course
+      const sourceCourse = await this.visitEndpoint({
+        path: `${API_PREFIX}/courses/${sourceCourseId}`,
+        action: 'get source course',
+        method: 'GET',
+      });
+      const applyAssignmentGroupWeights = sourceCourse.apply_assignment_group_weights;
+      // TODO: check if the assignment group name already exists in the destination course,
+      // in which we case we do not create a new assignment group
+      // instead, get the id of this matching assignment group and update weights if needed
+      // and also add this assignment group to the map
+      let destinationAssignmentGroup;
+      if (applyAssignmentGroupWeights) {
+        destinationAssignmentGroup = await this.api.course.assignmentGroup.create({
+          courseId: destinationCourseId,
+          name: sourceAssignmentGroup.name,
+          weight: sourceAssignmentGroup.group_weight,
+        });
+      } else {
+        destinationAssignmentGroup = await this.api.course.assignmentGroup.create({
+          courseId: destinationCourseId,
+          name: sourceAssignmentGroup.name,
+        });
+      }
+      // set apply_assignment_group_weights to true/false in the destination course
+      await this.visitEndpoint({
+        path: `${API_PREFIX}/courses/${destinationCourseId}`,
+        action: 'set apply_assignment_group_weights to true',
+        method: 'PUT',
+        params: {
+          course: {
+            apply_assignment_group_weights: applyAssignmentGroupWeights,
+          },
+        },
+      });
+
+      // add the mapping to the map
+      assignmentGroupMap[sourceId] = destinationAssignmentGroup.id;
+    }
+
+    // iterate through each source assignment
+    sourceAssignments.forEach(async (sourceAssignment: any) => {
+      // Get the assignment group id of the assignment
+      const assignmentGroupId = sourceAssignment.assignment_group_id;
+      const destinationAssignmentGroupId = assignmentGroupMap[assignmentGroupId];
+      // throw an error if the assignment group id is not in the map
+      if (!destinationAssignmentGroupId) {
+        throw new CACCLError({
+          message: 'Could not find assignment group id in map',
+          code: ErrorCode.MigrationIssue,
+        });
+      }
+      // determine the id of the assignment in the new course by using the map
+      const destinationAssignmentId = assignmentMap[sourceAssignment.id];
+      // throw an error if the assignment id is not in the map
+      if (!destinationAssignmentId) {
+        throw new CACCLError({
+          message: 'Could not find assignment id in map',
+          code: ErrorCode.MigrationIssue,
```

Comment:
For each `CACCLError`, create a new error code. It's okay for these to be very descriptive:

`ErrorCode.CouldNotFindDestinationAssignment` for example